packages/app/src/ui/pages/AddVerifiableCredential/useAddVerifiableCredential.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback } from "react";
+
+import { closePopup } from "@src/ui/ducks/app";
+import { useAppDispatch } from "@src/ui/ducks/hooks";
+import { addVerifiableCredential, rejectVerifiableCredentialRequest } from "@src/ui/ducks/verifiableCredentials";
+
+export interface IUseAddVerifiableCredentialData {
+  closeModal: () => void;
+  serializedVerifiableCredential: string;
+  onApproveAddVerifiableCredential: (verifiableCredentialName: string) => Promise<boolean>;
+  onRejectAddVerifiableCredential: () => void;
+}
+
+export const useAddVerifiableCredential = (): IUseAddVerifiableCredentialData => {
+  const { searchParams } = new URL(window.location.href.replace("#", ""));
+  const serializedVerifiableCredential = searchParams.get("serializedVerifiableCredential") || undefined;
```

Comment:
I think it makes more sense to treat a nonexistent value as undefined rather than null here, and would be more consistent with other parts of the code

Reply:
With typescript and `?.` operator we don't need to distinguish null and undefined. Moreover, it can be an issue when you only check the undefined value.